### PR TITLE
[Doc] Improve programming exercise instructor guide

### PR DESCRIPTION
**Pull request by kloessst**
### Motivation and Context

Improvements for the programming exercise documentation with focus on the exercise creation.

Changes can be checked here: https://artemis-platform.readthedocs.io/en/chore-programming-exercise-documentation/user/exercises/programming

### CODE REVIEW

**Review by JohannesWeiss — COMMENTED** [review]
For the supported SCA languages only Java is mentioned?
Shouldn't the Swift support be added there as well?

**Review by airelawaleria — COMMENTED** [review]
reviewed with @JohannesWeiss 

Also the link for sandbox in the end of the documentation can be displayed as a link instead of plain text

File: docs/user/exercises/programming-exercise-setup.rst [hunk 1]
```diff
@@ -187,28 +187,196 @@ Exercise Creation
 - However, if you have additional build steps or different configurations, you can adapt the BASE and SOLUTION build plan as needed
 - When students start the programming exercise, the current version of the BASE build plan will be copied. All changes in the configuration will be considered
 
-5. Adapt the interactive problem statement
+5. **Optional:** Configure static code analysis tools
+
+- The **Test** repository contains files for the configuration of static code analysis tools, if static code analysis was activated during the creation/import of the exercise
+- The folder *staticCodeAnalysisTools* contains configuration files for each used static code analysis tool
+- On exercise creation, Artemis generates a default configuration for each tool, which contain a predefined set of parameterized activated/excluded rules. The configuration files serve as a documented template that instructors can freely tailor to their needs.
```
Comment (derLalla) [inline docs/user/exercises/programming-exercise-setup.rst:194]:
```suggestion
- On exercise creation, Artemis generates a default configuration for each tool, which contains a predefined set of parameterized activated/excluded rules. The configuration files serve as a documented template that instructors can freely tailor to their needs.
```

File: docs/user/exercises/programming-exercise-setup.rst [hunk 2]
```diff
@@ -187,28 +187,196 @@ Exercise Creation
...
 - Click the |edit| button of the programming exercise or navigate into |edit-in-editor| and adapt the interactive problem statement.
 - The initial example shows how to integrate tasks, link tests and integrate interactive UML diagrams
 
-6. Configure Grading
+7. Configure Grading
+
+- **General Actions**
+
+  - |grading-save| Save the current grading configuration of the open tab
+  - |grading-reset| Reset the current grading configuration of the open tab to the default values. For **Test Case Tab**, all test cases are set to weight 1, bonus multiplier 1 and bonus points 0. For the **Cody Analysis Tab**, the default configuration depends on the selected programming language.
```
Comment (derLalla) [inline docs/user/exercises/programming-exercise-setup.rst:236]:
```suggestion
  - |grading-reset| Reset the current grading configuration of the open tab to the default values. For **Test Case Tab**, all test cases are set to weight 1, bonus multiplier 1 and bonus points 0. For the **Code Analysis Tab**, the default configuration depends on the selected programming language.
```

File: docs/user/exercises/programming-exercise-setup.rst [hunk 3]
```diff
@@ -187,28 +187,201 @@ Exercise Creation
...
-  .. figure:: programming/configure-grading.png
+    .. note::
+      **Example 3**: Given the configuration of **Example 2** with additional bonus points of 5 for TC B. The points achieved for passing TC A and TC C do not change. Passing TC B now accounts for 2.5 points plus 5 bonus points (7.5). If the student passes all test cases he will receive 10 (TC A) + 7.5 (TC B) + 2.5 (TC C) points, which amounts to 10 points and 10 bonus points and a score of 200%.
+
+  - On the right side of the page, charts display statistics about the current test case configuration. If changes are made to the configuration, a |grading-preview| of the statistics is shown.
+
+    - **Weight Distribution**: The distribution of test case weights. Visualizes the impact of each test case for the score calculation
+    - **Total Points**: The percentage of points given to students according to a specific test case. 100% in the chart represents full scores (100%) of **all** students
+
+- **Cody Analysis Tab**: Configure the visibility and grading of code quality issues on a category-level
```
Comment (derLalla) [inline docs/user/exercises/programming-exercise-setup.rst:283]:
```suggestion
- **Code Analysis Tab**: Configure the visibility and grading of code quality issues on a category-level
```

**Review by derLalla — CHANGES_REQUESTED** [review]
The text looks good. I have only left some changes

File: docs/user/exercises/programming.rst [hunk 4]
```diff
@@ -90,4 +90,4 @@ Its main features are
   or pointing to a possible location that caused an Exception
 * utilities to test exercises using System.out and System.in comfortably
 
-**For more information see https://github.com/ls1intum/artemis-java-test-sandbox**
+**For more information see `AJTS GitHub <https://github.com/ls1intum/artemis-java-test-sandbox>`__ **
```
Comment (FrankeLukas) [inline docs/user/exercises/programming.rst:93]: GitHub Actions complains here: `Inline strong start-string without end-string.`

Comment (FrankeLukas) [reply docs/user/exercises/programming.rst:93]:
```suggestion
**For more information see** `AJTS GitHub <https://github.com/ls1intum/artemis-java-test-sandbox>`__
```

File: docs/user/exercises/programming-exercise-setup.rst [hunk 5]
```diff
@@ -187,28 +187,201 @@ Exercise Creation
...
+  .. note::
+    For Swift, only the category Code Style can contain code quality issues currently. All other categories displayed on the grading page are dummies.
+
+  - On the left side of the page, instructors can configure the static code analysis categories.
+
+    - **Category**: The name of category defined by Artemis
+    - **State**:
+      - ``INACTIVE``: Code quality issues of an inactive category are not shown to students and do not influence the score calculation
+      - ``FEEDBACK``: Code quality issues of a feedback category are shown to students but do not influence the score calculation
+      - ``GRADED``: Code quality issues of a graded category are shown to students and deduct points according to the Penalty and Max Penalty configuration
```
Comment (FrankeLukas) [inline docs/user/exercises/programming-exercise-setup.rst:368]: This doesn't appear to work as intended:
grafik

**Review by FrankeLukas — APPROVED** [review]
Requests were addressed.

**Review by SimonEntholzer — APPROVED** [review]

**Review by derLalla — APPROVED** [review]
re-approving

**Review by airelawaleria — APPROVED** [review]

**Pull request merged**